src/Spryker/Client/RabbitMq/Model/Connection/ConnectionFactory.php

```diff
@@ -40,18 +40,29 @@ public function getQueueConnectionConfigs()
     }
 
     /**
-     * @param \Generated\Shared\Transfer\QueueConnectionTransfer $queueConnectionConfig
+     * @param \Generated\Shared\Transfer\QueueConnectionTransfer $queueConnectionTransfer
      *
      * @return \PhpAmqpLib\Connection\AMQPStreamConnection
      */
-    protected function createAmqpStreamConnection(QueueConnectionTransfer $queueConnectionConfig)
+    protected function createAmqpStreamConnection(QueueConnectionTransfer $queueConnectionTransfer)
     {
         $streamConnection = new AMQPStreamConnection(
-            $queueConnectionConfig->getHost(),
-            $queueConnectionConfig->getPort(),
-            $queueConnectionConfig->getUsername(),
-            $queueConnectionConfig->getPassword(),
-            $queueConnectionConfig->getVirtualHost()
+            $queueConnectionTransfer->getHost(),
+            $queueConnectionTransfer->getPort(),
+            $queueConnectionTransfer->getUsername(),
+            $queueConnectionTransfer->getPassword(),
+            $queueConnectionTransfer->getVirtualHost(),
+            $queueConnectionTransfer->getInsist() ?? false,
+            $queueConnectionTransfer->getLoginMethod() ?? 'AMQPLAIN',
+            $queueConnectionTransfer->getLoginResponse(),
+            $queueConnectionTransfer->getLocale() ?? 'en_US',
+            $queueConnectionTransfer->getConnectionTimeout() ?? 3,
+            $queueConnectionTransfer->getReadWriteTimeout() ?? 130,
+            null,
```

Comment:
is this always null?
vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/IO/StreamIO.php:59

Reply:
This looks like `$login_response`

Reply:
it's not always null, but we cannot have a "resource" in a transfer object